frontend/middleware_validatestatic.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-logr/logr"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+// Referenced in https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftresources
+var rxResourceGroupName = regexp.MustCompile(`^[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
+
+func MiddlewareValidateStatic(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+
+	subId := chi.URLParam(r, "subscriptionId")
+	resourceGroupName := chi.URLParam(r, "resourceGroupName")
+	resourceProviderNamespace := chi.URLParam(r, "resourceProviderNamespace")
+	resourceType := chi.URLParam(r, "resourceType")
+	operationId := chi.URLParam(r, "operationId")
+	resourceName := chi.URLParam(r, "resourceName")
+
+	apiVersion := r.URL.Query().Get(APIVersionKey)
+
+	if r.URL.Path != strings.ToLower(r.URL.Path) {
+		if log, ok := r.Context().Value(contextKeyOriginalPath).(logr.Logger); ok {
+			log.Error(errors.New("LowerCase error"), "path was not lower case")
+		}
+		arm.WriteError(w, http.StatusInternalServerError, arm.CloudErrorCodeInternalServerError, "", "Internal server error.")
+		return
+	}
+
+	if subId != "" {
+		valid := api.IsValid(subId)
+		if !valid {
+			arm.WriteError(w, http.StatusBadRequest, arm.CloudErrorCodeInvalidSubscriptionID, "", "The provided subscription identifier '%s' is malformed or invalid.", subId)
+			return
+		}
+	}
+
+	if resourceGroupName != "" {
+		if !rxResourceGroupName.MatchString(resourceGroupName) {
+			arm.WriteError(w, http.StatusBadRequest, arm.CloudErrorCodeResourceGroupNotFound, "", "Resource group '%s' is invalid.", resourceGroupName)
+			return
+		}
+	}
+
+	if resourceProviderNamespace != strings.ToLower(api.ProviderNamespace) {
+		arm.WriteError(w, http.StatusBadRequest, arm.CloudErrorCodeInvalidResourceNamespace, "", "The resource namespace '%s' is invalid.", resourceProviderNamespace)
+		return
+	}
+
+	if resourceType != "" {
+		if resourceType != strings.ToLower(api.ResourceType) {
+			arm.WriteError(w, http.StatusBadRequest, arm.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", resourceType, resourceProviderNamespace, apiVersion)
+			return
+		}
+	}
+
+	if resourceName != "" {
+		if !rxResourceGroupName.MatchString(resourceName) {
+			arm.WriteError(w, http.StatusBadRequest, arm.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s/%s' under resource group '%s' was not found.", resourceProviderNamespace, resourceType, resourceName, resourceGroupName)
```

Comment:
This doesn't actually check if the resourceName exists within a resourceGroup. If we are trying to validate the resourceName, we should have an error message tailored to the check. Do the resourceGroup and resourceName have the same requirements?

Reply:
Each resource has their own requirements. The resource group one is a good upper limit.  I can also narrow it down to this from the api spec:
            "name": "hcpOpenShiftClusterName",
            "in": "path",
            "description": "Name of HCP cluster",
            "required": true,
            "type": "string",
            "minLength": 3,
            "maxLength": 24,
            "pattern": "^[a-zA-Z0-9-]{3,24}$"
          }